drodrpg/DRODLib/Character.cpp

```diff
@@ -4306,25 +4301,24 @@ bool CCharacter::IsTileObstacle(
 //True if tile is an obstacle, false if not.
 const
 {
-	switch (GetResolvedIdentity())
+	switch (eMovement)
 	{
 		//These types can move through wall.
 		//NOTE: For greater scripting flexibility, these types will also be allowed
 		//to perform normal movement.
-		case M_SEEP:
+		case MovementType::WALL:
 		{
 			return CMonster::IsTileObstacle(wTileNo) &&
 				!(bIsWall(wTileNo) || bIsCrumblyWall(wTileNo) || bIsDoor(wTileNo));
 			//i.e. tile is considered an obstacle only when it blocks both movement types
 		}
 
 		//Flying types may also move over pits.
-		case M_WWING: case M_FEGUNDO:
+		case MovementType::AIR:
 			return CMonster::IsTileObstacle(wTileNo) &&
 					!bIsWater(wTileNo) && !bIsPit(wTileNo);
 
-		case M_WATERSKIPPER:
-		case M_SKIPPERNEST:
```

Comment:
Oh!  I didn't realize that M_SKIPPERNEST had water movement previously.  I must have misread the changes.  We should probably leave its movement type as-was, rather than changing it like I proposed.